src/core/mainloop/mainloop.c

```diff
@@ -2583,6 +2587,27 @@ write_bridge_ns_callback(time_t now, const or_options_t *options)
   return PERIODIC_EVENT_NO_UPDATE;
 }
 
+/**
+ * Periodic callback: if we're the bridge authority, set the running flag on
+ * bridges if they're reachable
+ */
+static int
+set_bridge_running_callback(time_t now, const or_options_t *options)
```

Comment:
Most of this function belongs in voteflags.c.

The callback-specific parts belong in this file, but the parts that update all the flags should be in voteflags.c.